hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -179,7 +180,10 @@ public void wipe(OwnershipTracker ownershipTracker, TokenRelationship accountRel
 
 		validateTrue(serialNumbers.size() > 0, FAIL_INVALID, () ->
 				"Cannot wipe " + serialNumbers.size() + " number of Unique Tokens.");
-
+		for (UniqueToken token : loadedUniqueTokens) {
+			validateTrue(token.getOwner().equals(accountRel.getAccount().getId()), FAIL_INVALID, () ->
+					"Cannot wipe tokens which given account does not own.");
+		}
```

Comment:
I think that `serialNumbers` argument and the `loadedUniqueTokens` are kind of separated at the moment. We are relying on the token transition model to have loaded all unique tokens necessary. I would prefer if we either:
- remove `List<long> serialNumbers` and iterate only through loadedUniqueTokens to remove them - this however is kind of shady if you do not know that you will wipe the `loadedUniqueTokens`
- change `loadedUniqueTokens` to `Map` where the key is the `serialNumber`, then when we call `wipe` we iterate through all `serialNumbers`, get the `uniqueToken` from the map and verify the owner. 


I think that the second option is the best one